[9.0] medor on tour - product subscription

robinkeunen opened the pull request.

robinkeunen committed:
[ADD] product_subscription_web_access module

robinkeunen committed:
[IMP] ps: add demo data

houssine78 committed:
[ADD] add module website_product_suscbrption_online_payment

[IMP] make var name shorter

[IMP] fix indent

houssine78 committed:
[IMP] refact to allow to be overridden

[IMP] make it generic

moving mollie related code to a dedicated module

[IMP] dedicated thanks page

override mollie redict to allows to have a dedicated thanks page for
product subscription after Mollie payment validation

[IMP] remove commented code

[TODO] add todo

[IMP] refact to allow to be overridden

[IMP] make it generic

moving mollie related code to a dedicated module

houssine78 committed:
[IMP] dedicated thanks page

override mollie redict to allows to have a dedicated thanks page for
product subscription after Mollie payment validation

robinkeunen committed:
[FIX] ps_web_access: is_web_subscribed computation

robinkeunen committed:
[REF] ps: start_date replaces subscribed_on

robinkeunen committed:
[FIX] wps: signup email rather than reset password

robinkeunen committed:
[IMP] ps: name not required on subscriptions and state clickable

houssine78 committed:
[FIX] fix effective date setting

[FIX] add missing import

robinkeunen committed:
[REF] wps: minor controller refactoring

houssine78 committed:
[IMP] add specific redirection for transaction state

[FIX] resolve conflicts

houssine78 committed:
[IMP] put template ids in local static variable

[FIX] resolve conflict

[FIX] resolve conflict

[IMP] remove commented code

houssine78 committed:
[IMP] fix routes

houssine78 committed:
[IMP] add related transaction state on subscription request

[IMP] add string on the related field to don't mix up with status

robinkeunen committed:
Merge branch '9.0-medor-web-subscription' into 9.0-medor-on-tour-ps

robinkeunen committed:
Merge branch '9.0_enable_online_payment' into 9.0-medor-on-tour-ps

robinkeunen committed:
Merge branch '9.0-medor-renew-subscription' into 9.0-medor-on-tour-ps

robinkeunen committed:
[IMP] wps: add form header text

robinkeunen committed:
[REF] wps_mollie,online: group form fields

robinkeunen committed:
[IMP] ps_web_access: add web access information section

robinkeunen committed:
[IMP] wps: remove useless close button from ps form

robinkeunen committed:
[REF] ps: split models into separate files

robinkeunen committed:
fixup! [REF] ps: split models into separate files

robinkeunen committed:
[FIX] ps_web_access: partner has access to web even when state is terminated (when last product was released)

robinkeunen committed:
[REF] ps: always set sponsor, sponsor always pays. subscription stypes

robinkeunen committed:
[IMP] wps: redirect param in become_subscriber form

robinkeunen commented on product_subscription/models/invoice.py:
```diff
@@ -56,16 +81,17 @@ def confirm_paid(self):
                     move_line = invoice.payment_move_line_ids[0]
                     effective_date = move_line.date
 
-                invoice.process_subscription(effective_date)
```
@houssine this goes through post_process_confirm_paid defined in `easy_my_coop.models.account_invoice.py` it therefore calls set_cooperator_effective and creates a cooperator share line at line 78

houssine78 replied:
I see... in fact it should go back to invoice.process_subscription(effective_date) as process_subscription function is not defined in easy my coop. Did I did this ?

robinkeunen replied:
I think you did this yes (_git blame is watching you_) :p The other option is to set the invoice as `cooperator_share = True` like you do for subscriptions. Can you fix it today or should I do it?

houssine78 replied:
ok done. I finally changed the function name as these functions don't rely on each other. this is the confirm paid function that handle the process and hence should call the right function

robinkeunen committed:
[IMP] solve warning and make installable from scratch

robinkeunen committed:
[IMP] ps_web_access: give web access to open requests

remytms committed:
[REF] website_product_subscription: New subscription request model

houssine78 committed:
[FIX] change post process function name

To avoid collision with a function defined with  in easy my coop and to
complexify the function handling by overriding it we just change the
function name as they don't rely on each other

remytms committed:
[REF] w_p_s_online_payment: Adapt to new subscription form

remytms committed:
[REF] p_s_web_access: Adapt to new subscription form

robinkeunen commented on website_product_subscription/controllers/subscribe.py:
```diff
@@ -0,0 +1,268 @@
+        [
+            # '/page/become_subscriber',
+            # '/become_subscriber',
+            '/new/subscription',
```
@remytms You should grep `become_subscriber`, there are still uses of this route in the codebase.

remytms replied:
@robinkeunen Yes, but I'm not sure of removing `/become_subscriber` page. For now, I prefer giving the new url to Médor. And keep the old one to keep compatibility with the old behaviour. Because, medor will perhaps change the default behavour of this form.

robinkeunen replied:
Ok to keep the old one. Don't forget to communicate the new routes to Jonathan.

robinkeunen commented on website_product_subscription/controllers/subscribe.py:
```diff
@@ -0,0 +1,268 @@
+        auth='public',
+        website=True
+    )
+    def subscribe(self, **kwargs):
```
@remytms  
trop bien 👍

robinkeunen commented on website_product_subscription/controllers/subscribe.py:
```diff
@@ -0,0 +1,268 @@
+        )
+
+    @http.route(
+        '/new/gift/subscription',
```
@remytms  for  consistency, it should be
- `/new/subscription/gitf`
and the other 
- `/new/subscription/basic` 

or even `susbcription/create` (ressource than REST verb)

remytms replied:
@robinkeunen Just THANKS ! I will reword URL this way. :D

remytms committed:
[CHG] website_p_s: Route names

Leave `become_subscriber`, and rename routes for subscriptions.

remytms committed:
[REF] w_p_s_online_payment: Keep compatibility

remytms committed:
[ADD] website_p_s: Return button on success page

remytms committed:
[ADD] p_s_web_access: New webaccess field

remytms committed:
[IMP] website_p_s: Let overwrite subscription request creation

remytms committed:
fixup! [REF] website_product_subscription: New subscription request model

remytms committed:
[ADD] p_s_web_access: webaccess in new subscription form

robinkeunen committed:
[IMP] ps: add _descriptiond to models

robinkeunen committed:
[ADD] ps_mass_mailing: new module

remytms committed:
fixup! [REF] website_product_subscription: New subscription request model

remytms committed:
fixup! [ADD] p_s_web_access: webaccess in new subscription form

remytms committed:
[ADD] website_p_s: Back button on form

robinkeunen committed:
[IMP] ps: add renewed boolean field

houssine78 committed:
[FIX] fix in case of redirect is empty

houssine78 committed:
[IMP] add split payment option

houssine78 committed:
[IMP] apply split payment price

if a product subscription template with split payment True is selected
then send the split payment price as amount to pay trough online payment

houssine78 committed:
[IMP] send the invoice when split payment price is true

houssine78 committed:
[FIX] in case of lase.end_date is not defined

remytms committed:
[FIX] p_s_web_access: Content of web access text info

remytms committed:
[FIX] website_p_s: Recaptcha validation

robinkeunen committed:
fixup! [IMP] ps: add renewed boolean field

robinkeunen committed:
fixup! [ADD] ps_mass_mailing: new module

houssine78 committed:
Merge pull request #10 from coopiteasy/9.0-split-payment-payment

9.0 split payment

houssine78 committed:
[IMP] remove redirection from here

remytms committed:
[FIX] website_p_s: Email recording

robinkeunen committed:
[IMP] hide and display presentation text

robinkeunen committed:
[ADD] ps_mass_mailing: add cron and active field

remytms committed:
[FIX] medor_custom: Email in new subscription form

remytms committed:
fixup! [FIX] website_p_s: Recaptcha validation

robinkeunen committed:
[ADD] confition générales de ventes

houssine78 committed:
[IMP] also handle transaction creation when payment type is deferred

remytms committed:
[FIX] website_p_s: Subscription form when user is connected

remytms committed:
[REF] p_s_web_access: Rename webaccess to web subscriber

houssine78 committed:
[IMP] authorize split payment thru payment transfer

so we also activate the subscription when on transfer payment when split
payment subscription

robinkeunen committed:
[FIX] ps_online: fix post_process_confirm_sub_paid for diferred payment

houssine78 committed:
[REV] revert to previous commit

meanwhile i've added the transaction creation in the copy/pasted
function (which is bad practice by the way)

remytms committed:
[ADD] website_p_s: Login warning banner in subscription form

robinkeunen committed:
[FIX] get template on request and not on invoice

robinkeunen committed:
[ADD] website_ps: add sale conditions on gift form

houssine78 committed:
[IMP] get the presentation from the selected subscription template

robinkeunen committed:
[FIX] website_ps: conditions required

houssine78 committed:
[IMP] if split payment is choosen it should not be flagged as paid

houssine78 committed:
[IMP] payment confirmation mail shouldn't be sent when split payment

houssine78 committed:
[FIX] add missing import

robinkeunen committed:
[ADD] ps_web_access: add presention text in template

robinkeunen committed:
fixup! [ADD] ps_web_access: add presention text in template

houssine78 committed:
[FIX] action_reset_password() must be call on a record not on the pool

remytms committed:
[FIX] website_p_s: From not initialized when POST method

robinkeunen committed:
[ADD] ps_web_access: temporary_access_length configuration view

houssine78 committed:
[IMP] recall send invoice after the transaction has been created

remytms committed:
[FIX] website_p_s: gift flag on subscription request

robinkeunen committed:
[FIX] wps: error in filling recaptcha renders a filled form

remytms committed:
[FIX] website_p_s: Correctly write invoice address and VAT
